packages/vscode-extension/src/debugging/RadonCDPProxyDelegate.ts

```diff
@@ -26,8 +29,32 @@ export class RadonCDPProxyDelegate implements CDPProxyDelegate {
         return this.handleConsoleAPICalled(applicationCommand);
       }
       case "Debugger.paused": {
+        if (this.resumeEventTimeout) {
+          clearTimeout(this.resumeEventTimeout);
+        }
         return this.handleDebuggerPaused(applicationCommand, tunnel);
       }
+      case "Debugger.resumed": {
+        if (this.justCalledStepOver) {
+          // when step-ober is called, we expect Debugger.resumed event to be called
+          // after which the paused event will be fired almost immediately as the
+          // debugger stops at the next line of code.
+          // In order to prevent the paused event from being fired immediately resulting
+          // in the overlay blinking for a fraction of second, we wait for a short period
+          // just in case the paused event is never fired.
+          this.justCalledStepOver = true;
+          if (this.resumeEventTimeout) {
+            clearTimeout(this.resumeEventTimeout);
+          }
+          this.resumeEventTimeout = setTimeout(() => {
+            this.justCalledStepOver = false;
+            this.debuggerResumedEmitter.fire({});
+          }, 100);
+        } else {
+          this.debuggerResumedEmitter.fire({});
+        }
+        return applicationCommand;
```

Comment:
I'd extract this into a separate `handleResumed` method to reduce the clutter here.